internal/core/domain/role_permissions.go

```diff
@@ -0,0 +1,6 @@
+package domain
+
+type RolePermissions struct {
+	RoleID  string   `json:"role_id"`
+	PermIDs []string `json:"perm_id"`
```

Comment:
Please add database reference constraints so that it will restrict you from save wrong ids that does exist on either role or permission table. define the relationship between all the tables

Reply:
This has already been done by the database at creation of the tables

Reply:
The relationship between all the tables is impossible to do as I am not using gorm